src/creature.cpp

```diff
@@ -448,8 +449,8 @@ CreatureVector Creature::getKillers()
 	const int64_t timeNow = OTSYS_TIME();
 	const uint32_t inFightTicks = getNumber(ConfigManager::PZ_LOCKED);
 	for (const auto& it : damageMap) {
-		Creature* attacker = g_game.getCreatureByID(it.first);
-		if (attacker && attacker != this && timeNow - it.second.ticks <= inFightTicks) {
+		std::shared_ptr<Creature> attacker = g_game.getCreatureByID(it.first);
```

Comment:
Could this be `const auto`
```suggestion
		const auto attacker = g_game.getCreatureByID(it.first);
```